fastparquet/writer.py

```diff
@@ -737,11 +788,19 @@ def write(filename, data, row_group_offsets=50000000,
         resolution; in "int96" mode, they are written as 12-byte blocks, with
         the first 8 bytes as ns within the day, the next 4 bytes the julian day.
         'int96' mode is included only for compatibility.
+    quantization_level: int or None
+        If not None, then quantization_level floating point values in this many units of the
+        median absolute deviation of the column. Only works for the 'simple' file
+        scheme.
 
     Examples
     --------
     >>> fastparquet.write('myfile.parquet', df)  # doctest: +SKIP
     """
+
+    global COLUMN_MADS
+    COLUMN_MADS = {}
```

Comment:
Just to cache stats about the data. The computation of the scale is a real bottle neck. It is cached here and then written directly in the compressed format. This is only a performance thing. It was easiest to add it in this way, but yes, this implementation of the caching is ugly.